<?xml version="1.0"?>

<launch>
  <arg name="map_file" default="$(find orne_navigation_executor)/maps/mymap.yaml"/>
  <arg name="waypoints_file" default="$(find orne_navigation_executor)/waypoints_cfg/waypoints_beta.yaml"/>
  <arg name="save_joy_button" default="0"/>
  <arg name="joy_config" default="elecom_joy"/>
  <arg name="joy_dev" default="/dev/input/js0"/>

  <include file="$(find icart_mini_driver)/launch/teleop_joy.launch">
    <arg name="joy_config" value="$(arg joy_config)"/>
    <arg name="joy_dev" value="$(arg joy_dev)"/>
  </include>

  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>

  <node pkg="orne_waypoints_editor" name="orne_waypoints_editor" type="orne_waypoints_editor" output="screen">
    <param name="filename" value="$(arg waypoints_file)"/>
    <param name="world_frame" value="map"/>
    <remap from="waypoints_viz" to="clicked_point"/>
    <remap from="finish_pose" to="move_base_simple/goal"/>
    <param name="save_joy_button" value="$(arg save_joy_button)"/>
    <remap from="waypoints_joy" to="joy"/>
  </node>

  <include file="$(find orne_navigation_executor)/launch/localization_beta.launch"/>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find orne_waypoints_editor)/rviz_cfg/orne_waypoints_editor.rviz"/>
</launch>

